### PR TITLE
Modify create_updated_flutter_deps script to create a monorepo DEPS

### PR DESCRIPTION
**Pull request by whesse**
The script used to roll changes in Dart's third-party dependency version pins to Flutter engine is given a command-line
option to create a new, modified DEPS file.

This DEPS file is used by a gclient solution that can check out specified versions of Dart, engine, and framework in the same
gclient solution.  This is a prototype of a monorepo that can be used to build and test tip-of-tree Dart and Flutter engine together,
and is hosted at https://dart.googlesource.com/monorepo

### CODE REVIEW

File: tools/dart/create_updated_flutter_deps.py [hunk 1]
```diff
@@ -61,12 +62,29 @@ def ParseArgs(args):
       type=str,
       help='Flutter DEPS file.',
       default=os.path.join(FLUTTER_ROOT, 'DEPS'))
+  parser.add_argument('--monorepo_deps', '-m',
+      type=str,
+      help='Monorepo DEPS file.',
+      default=os.path.join(MONOREPO_ROOT, 'DEPS'))
+  parser.add_argument('--create_monorepo',
+                      action='store_true',
```
Comment (zanderso) [inline tools/dart/create_updated_flutter_deps.py:70]: nit: use indentation consistent with the other `add_argument` calls.

File: tools/dart/create_updated_flutter_deps.py [hunk 2]
```diff
@@ -98,7 +126,36 @@ def Main(argv):
         updatedfile.write("  '%s': '%s',\n" % (k, v))
       updatedfile.write('\n')
 
+    elif (create_monorepo
+       and lines[i].startswith('  # Download a prebuilt Dart SDK by default')
+       and lines[i+1].startswith("  'download_dart_sdk': True,")):
+        sdk_download = True
```
Comment (zanderso) [inline tools/dart/create_updated_flutter_deps.py:132]: nit: This file appears to be using a two space indent.

File: tools/dart/create_updated_flutter_deps.py [hunk 3]
```diff
@@ -98,7 +126,36 @@ def Main(argv):
...
+        updatedfile.write("  'download_dart_sdk': False,\n")
+        i = i + 1
+
+    elif (create_monorepo
+          and lines[i].startswith('deps = {')
+          and lines[i+1].startswith(
+          "  'src': 'https://github.com/flutter/buildroot.git' + '@' + ")):
+        pinned_monorepo_deps = True
```
Comment (zanderso) [inline tools/dart/create_updated_flutter_deps.py:140]: ditto

File: tools/dart/create_updated_flutter_deps.py [hunk 4]
```diff
@@ -133,9 +190,25 @@ def Main(argv):
...
-  os.rename(updatedfilename, args.flutter_deps)
-
+  os.remove(output_file)
+  os.rename(updatedfilename, output_file)
+
+  if create_monorepo and not (vars_updated and
+      sdk_download and
+      dart_pin and
+      pinned_monorepo_deps and
+      updated_dependencies ):
```
Comment (zanderso) [inline tools/dart/create_updated_flutter_deps.py:200]: nit: extra space after `updated_dependencies`.

File: tools/dart/create_updated_flutter_deps.py [hunk 5]
```diff
@@ -98,7 +126,36 @@ def Main(argv):
...
+        i = i + 1
+        updatedfile.write(lines[i])
+        for dependency in [('src/flutter', 'flutter_git', '/mirrors/engine', args.engine_commit),
+                           ('src/third_party/dart', 'dart_git', '/sdk', args.dart_commit),
+                           ('flutter', 'flutter_git', '/mirrors/flutter', args.flutter_commit),
+                           ]:
+          updatedfile.write("\n  '%s': Var('%s') + '%s' + '@' + '%s',\n" % dependency)
+
+    elif ( create_monorepo
+           and i < len(lines) - 3
```
Comment (zanderso) [inline tools/dart/create_updated_flutter_deps.py:150]: nit: inconsistent indentation.

**Review by zanderso — CHANGES_REQUESTED** [review]

**Comment by whesse** [comment]
Based on review comments, I will take a different direction with this.
The changes to make the monorepo DEPS will be performed by a different script, that will run after the create_updated_flutter_deps script (which will not be modified)

The prototype script here will be copied to the monorepo directory, and used there.

**Pull request closed without merging**